src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -457,7 +458,18 @@ func (r *StatefulSetResource) obj() (k8sclient.Object, error) {
 	return ss, nil
 }
 
-func overprovisioned(developerMode bool, limits corev1.ResourceList) []string {
+func prepareAdditionalArguments(
+	developerMode bool, limits, requests corev1.ResourceList,
+) []string {
+	var smp int64 = 1
+	if !limits.Cpu().IsZero() {
+		limits.Cpu().RoundUp(0)
```

Comment:
`Roundup` modifies the underlying resource, I don't think we should modify a user request, even though integer CPU requests (and no limit) is recommended. I think `DeepCopy()` is needed.

Reply:
Yeah, certainly. Updated.